components/EntireScheme.svelte

```diff
@@ -82,12 +83,22 @@
   }
 
   function backfill(json) {
-    // Look for any LineStrings without length_meters. Old route-snapper versions didn't fill this out.
+    let idCounter = 0;
     for (let f of json.features) {
+      // Look for any LineStrings without length_meters. Old route-snapper versions didn't fill this out.
       if (f.geometry.type == "LineString" && !f.properties.length_meters) {
         f.properties.length_meters =
           length(f.geometry, { units: "kilometers" }) * 1000.0;
       }
+      // TODO Hack around https://github.com/mapbox/mapbox-gl-js/issues/2716.
```

Comment:
This is kind of a nightmare. I just want every feature to have a unique ID, but we can't do something simple like use a `uuid` library, because parts of maplibre **silently** break on non-numeric IDs. But using numeric IDs like this is also confusing:

- this backfill has a bug and might double-assign an ID, the way it's written right now
- `0` acts like `false` in JS silently, and I've been bitten by `if (id)` instead of `id (id != null)` so many times
- mapbox-gl-draw lets us use our own feature IDs... unless it's `0`!